covid19-dashboard/dc-covid19/src/MetaDataGenerator.ts

```diff
@@ -1,123 +1,178 @@
+/**
+ Copyright 2020 Google LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ https://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
 
-type Metadata = {
-    geoId: string,
-    name: string,
-    onHoverInfo: string[]
-    textOnTopOfBar: string,
-    value: number
-}
+/**
+ * Given two values, return the result of calculationType.
+ * @param values: values[0] is date - deltaDays, values[1] is the date.
+ * @param calculationType: the type of calculating we are performing
+ */
 
-type DateToGeoIdToValue = {string: {string: number}} | {}
+import {numberWithCommas} from "./Utils";
+
+type DateToGeoIdToValue = {[geoId: string]: {[date: string]: number}} | {}
 
 /**
- * Given a list of parameters, return human readable strings explaining the data.
+ * Returns strings briefly explaining the calculation performed
  * @param geoId: the geoId of the region
  * @param regionName: the name of the region
  * @param date: the date of the observation
  * @param value: the value at that specific date for that specific geoId
  * @param population: the total population of that region
- * @param absoluteIncreaseValue: the absolute difference between day 2 and day 1
+ * @param absolute: the absolute difference between day 2 and day 1
  * @param label: Are we talking about cases? or deaths?
- * @param contentType: What is the graph showing? Per-Capita? Increase? Absolute?
+ * @param calculationType: calculations performed on data: eg. ['difference']
+ * @return onHoverInfo: human readable text to display on graph's on-hover
  */
-const getOnHoverInfoText = (geoId: string, regionName: string, date: string, value: number, population: number,
-                             absoluteIncreaseValue: number, label: string, contentType: string) => {
-    let disclaimer: string = ""
-    let onHoverInfo: string[];
-    // If it's New York City or Kansas City. Show disclaimer. These are NYT exceptions.
-    if (geoId === 'geoId/3651000' || geoId === 'geoId/2938000')
-        disclaimer = `All counties in ${regionName} are combined and reported as one.`
-
-    if (contentType === 'perCapita' || contentType === 'absolutePerCapita') {
+const getOnHoverInfoText = (geoId: string,
+                            regionName: string,
+                            date: string,
+                            value: number,
+                            population: number,
+                            absolute: number,
+                            label: string,
+                            calculationType: string[]) => {
+    let disclaimer = ""
```

Comment:
Please add semicolon: https://google.github.io/styleguide/jsguide.html#formatting-semicolons-are-required

for all statements